db/init_db.js

```diff
@@ -1,31 +1,380 @@
 // code to build and initialize DB goes here
-const {
-  client
-  // other db methods 
-} = require('./index');
+const client = require("./client");
+const { createOrder, getAllOrders } = require("./orders");
+const { createUser } = require("./users");
+const { createProduct, getAllProducts, getProductByID } = require("./products");
+const dotenv = require("dotenv")
+dotenv.config()
+
+// const { createUser } = require("./db");
+const { createProduct } = require("./products");
+
+async function dropTables() {
+  try {
+    console.log("Dropping Tables ...");
+    await client.query(`
+      DROP TABLE IF EXISTS order_products;
+      DROP TABLE IF EXISTS orders;
+      DROP TABLE IF EXISTS products;
+      DROP TABLE IF EXISTS users;
+    `);
+    console.log("Finished dropping tables.");
+  } catch (error) {
+    throw error;
+  }
+}
 
 async function buildTables() {
   try {
-    client.connect();
+    console.log("Building Tables ...");
+    await client.query(`
+      CREATE TABLE users(
+        id SERIAL PRIMARY KEY,
+        "firstName" VARCHAR(255) NOT NULL,
+        "lastName" VARCHAR(255) NOT NULL,
+        email VARCHAR(255) UNIQUE NOT NULL,
+        "imageURL" VARCHAR(255) DEFAULT 'imageUrl',
+        username VARCHAR(255) UNIQUE NOT NULL,
+        password VARCHAR(255) NOT NULL,
+        "isAdmin" BOOLEAN DEFAULT false NOT NULL
+      );
+      CREATE TABLE products(
+        id SERIAL PRIMARY KEY,
+        name VARCHAR(255) UNIQUE NOT NULL,
+        description VARCHAR(255) NOT NULL,
+        price NUMERIC(7, 2),
+        "imageURL" VARCHAR(255) DEFAULT 'imageUrl',
+        "inStock" BOOLEAN DEFAULT false NOT NULL,
+        category VARCHAR(255) NOT NULL
+      );
+      CREATE TABLE orders(
+        id SERIAL PRIMARY KEY,
+        status VARCHAR(255) DEFAULT 'created',
+        "userId" INTEGER REFERENCES users(id),
+        "datePlaced" DATE NOT NULL DEFAULT CURRENT_DATE
+      );
+      CREATE TABLE order_products(
+        id SERIAL PRIMARY KEY,
+        "productId" INTEGER REFERENCES products(id) UNIQUE,
+        "orderId" INTEGER REFERENCES orders(id) UNIQUE,
+        price NUMERIC(7, 2),
+        quantity INTEGER NOT NULL DEFAULT 0
+      );
+    `);
+    console.log("Finished building tables.");
+  } catch (error) {
+    throw error;
+  }
+}
+
+//map calls the callback func once for each element (good), in order (not necessary),
+//+ constructs new array from results
+
+async function createInitialUsers() {
+  try {
+    console.log("Creating Users ...");
+    // create useful starting data
+    await createUser({
+      username: "beatricewhite",
+      password: "111111",
+      firstName: "Beatrice",
+      lastName: "White",
+      email: "beatricewhitee@gmail.com",
+      isAdmin: false,
```

Comment:
instead of awaiting each function call individually, you can create a list of objects (i.e. users) and the pass that to Promise.all(), calling the function on each object.

```
const users = [{id: 1}, {id: 2}, {id: 3})];
const users = await Promise.all(users.map(user => createUser(user)));
```